tests/src/rules/no-extraneous-dependencies.js

```diff
@@ -128,3 +128,30 @@ ruleTester.run('no-extraneous-dependencies', rule, {
     }),
   ],
 })
+
+ruleTester.run('no-extraneous-dependencies core-module', rule, {
+  valid: [
+    test({
+      code: 'import "core-module"',
+      settings: { 'import/core-modules': ['core-module'] },
+    }),
+    test({
+      code: 'import "@namespaced/module"',
+      settings: { 'import/core-modules': ['@namespaced/module'] },
+    }),
+    test({
+      code: 'import "@namespaced/module/and/sub/path"',
```

Comment:
I'm not sure if this is supposed to work, but I was trying to use `core-modules` to fix the `no-extraneous-dependencies` for my webpack `resolve.alias` that points `@namespaced/module/lib/foo` to `./src/foo`

Maybe related to #479 ?